SpatialGDK/Source/SpatialGDK/Private/Utils/ActorGroupManager.cpp

```diff
@@ -0,0 +1,202 @@
+#include "Utils/ActorGroupManager.h"
+#include "SpatialGDKSettings.h"
+
+UActorGroupManager::UActorGroupManager()
+{
+	InitFromSettings();
+}
+
+UActorGroupManager* UActorGroupManager::GetInstance()
+{
+	UActorGroupManager* Manager = NewObject<UActorGroupManager>();
+	Manager->InitFromSettings();
+	return Manager;
+}
+
+void UActorGroupManager::InitFromSettings()
+{
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+
+	ClassPathToActorGroup.Empty();
+	ActorGroupToWorkerType.Empty();
+
+	auto ActorGroups = DefaultActorGroups();
+	auto WorkerAssociation = DefaultWorkerAssociation();
+	if (Settings->bEnableOffloading)
+	{
+		ActorGroups = Settings->ActorGroups;
+		WorkerAssociation = Settings->WorkerAssociation.ActorGroupToWorker;
+	}
+
+	for (TPair<FName, FActorClassSet> ActorGroup : ActorGroups)
+	{
+		for (TSoftClassPtr<AActor> ClassPtr : ActorGroup.Value.ActorClasses)
+		{
+			ClassPathToActorGroup.Add(ClassPtr, ActorGroup.Key);
+		}
+	}
+
+	ActorGroupToWorkerType.Append(WorkerAssociation);
+}
+
+FName UActorGroupManager::GetActorGroupForClass(UClass* Class)
+{
+	if (Class == nullptr || !Class->IsChildOf<AActor>())
```

Comment:
What's the justification for checking this here vs. just having AActor as the param?